numpy/array_api.py

```diff
@@ -0,0 +1,24 @@
+import enum
+
+__all__ = [
+    'CopyMode'
+    ]
+
+class CopyMode(enum.Enum):
+
+    ALWAYS = True
+    IF_NEEDED = False
+    NEVER = 2
+
+    def __bool__(self):
+        # For backwards compatiblity
+        if self == CopyMode.ALWAYS:
+            return True
+
+        if self == CopyMode.IF_NEEDED:
+            return False
+
+        raise TypeError(f"{self} is neither True nor False.")
+
+
+CopyMode.__module__ = 'numpy.array_api'
```

Comment:
@rgommers @seberg @mattip Is this change good? I shifted `CopyMode` to `numpy.array_api` after discussing with @rgommers

Reply:
Sorry, but no, that does not make sense to me.  You want this for that maybe, but it will effectively be internal/semi-private NumPy API and thus has to live in NumPy proper.  `numpy.array_api` is not NumPy proper, for one, we can't import it since it will get a warning (probably).

Reply:
I see. Actually, the original conclusion that I am aware of is, "keep using `CopyMode` enum but keep them private". This is to avoid pollution of public `numpy` namespace but still take advantage of an enum. Though if `CopyMode` is kept private then how will a user access and pass it as input to functions like `numpy.asarray`. So, we decided to use `numpy.array_api` (discussed in [NEP 47](https://numpy.org/neps/nep-0047-array-api-standard.html)) as there no concerns of backwards compatibility and pollution of namespace with it. 
Let me know if there is an alternative to make this better. Thanks.

cc @rgommers @shoyer @asmeurer

Reply:
With "keep it private" I believe the intent was simply `numpy.CopyMode` -> `numpy._CopyMode` (with underscore). And then merge this. Using `_CopyMode` from the `array_api` module is a separate step/PR.

Reply:
I see. I will send the updates here for making `CopyMode` public.

Reply:
I have made the change. Please let me know if any changes are required.